### PR TITLE
Chore/146687435/stabilise base application

### PR DESCRIPTION
**Pull request by joel-ace**
### What does this PR do?
fix dependency issues so app can be deployed on heroku without errors
fix merge conflicts

### Description of Task to be completed?
None

### How should this be manually tested?
None

### Any background context you want to provide?
None

### What are the relevant pivotal tracker stories?
None

### Screenshots (if appropriate)
None

### Questions:
None

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-MONGOHQ_URL=localhost
+MONGOHQ_URL=mongodb://cfh:qwerty1234@ds123752.mlab.com:23752/haku-cfh
```
Comment (tonerdo) [inline .env]: DO NOT put service credentials in a public repo

File: .env [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 FB_CLIENT_ID=
 FB_CLIENT_SECRET=
 TWITTER_CONSUMER_KEY=
```
Comment (tonerdo) [inline .env]: It's either you give env keys values or don't put them at all

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,7 @@ bower_components
 .env
 coverage
 npm-debug.log
+<<<<<<< HEAD
+.env
+=======
+>>>>>>> 7278826f71864b65073709d7ff0e018f5d94dd59
```
Comment (tonerdo) [inline .gitignore]: Why was a merge conflict marker commited?

File: config/env/all.js [hunk 4]
```diff
@@ -5,5 +5,5 @@ var keys = rootPath + '/keys.txt';
 module.exports = {
 	root: rootPath,
 	port: process.env.PORT || 3000,
-    db: process.env.MONGOHQ_URL
+    db: process.env.MONGOLAB_URL
```
Comment (tonerdo) [inline config/env/all.js]: You env file says `MONGOHQ_URL` but this says MONGOLAB_URL

File: sample.env [hunk 5]
```diff
@@ -0,0 +1,7 @@
+MONGOHQ_URL=
+FB_CLIENT_ID=
+FB_CLIENT_SECRET=
+TWITTER_CONSUMER_KEY=
+TWITTER_CONSUMER_SECRET=
+GITHUB_CLIENT_ID=
+GITHUB_CLIENT_SECRET=
```
Comment (tonerdo) [inline sample.env]: It's either you give env keys values or don't put them at all. What exactly does this file do?

**Review by tonerdo — CHANGES_REQUESTED** [review]
A bunch of things you need to look at. Also your PR description isn't well formatted

File: sample.env (hunk 5 above)
Comment (joel-ace) [reply sample.env]: The file contains environment variables associated with this project. This is to ensure all team members sync while working on different features of the project as more than one team member can be assigned to a particular task

File: config/env/all.js (hunk 4 above)
Comment (tonerdo) [reply config/env/all.js]: You need a default value for `db` much like you have for `port`

**Review by tonerdo — CHANGES_REQUESTED** [review]

File: config/env/all.js [hunk 6]
```diff
@@ -5,5 +5,5 @@ var keys = rootPath + '/keys.txt';
 module.exports = {
 	root: rootPath,
 	port: process.env.PORT || 3000,
-    db: process.env.MONGOHQ_URL
+    db: process.env.MONGOLAB_URL || 'localhost',
```
Comment (tonerdo) [inline config/env/all.js]: This is very wrong and will not work. Where's the `mongodb://` prefix before `localhost`? Kindly look up how to construct a valid mongodb connection url

**Review by tonerdo — CHANGES_REQUESTED** [review]

**Review by tonerdo — APPROVED** [review]

**Pull request merged**